### PR TITLE
[SCB-1044]add current process CPU rate  and net packets in the metrics

### PR DESCRIPTION
**Pull request by heyile**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java [hunk 1]
```diff
@@ -18,68 +18,126 @@
...
     rate = 0.0;
+    processRate = 0.0;
+  }
+
+  private String getCurrentPid() {
+    String name = ManagementFactory.getRuntimeMXBean().getName();
+    if (name.contains("@")) {
+      return name.substring(0, name.indexOf("@"));
+    }
+    LOGGER.warn("Failed to get current process id. {}", name);
```
Comment (liubao68) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: If PID is null, we should stop to calculate data. 
BTW, when will pid is null?

Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: do not know when will pid is null. I's impossible that the pid is null.

Comment (liubao68) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: If it is impossible, you can throw an IllegalStateException here.

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/ClientEndpointsLogPublisher.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public ClientEndpointsLogPublisher(MeasurementTree tree, StringBuilder sb, Strin
   @Override
   public void print(boolean printDetail) {
     appendLine(sb, "    client.endpoints:");
-    appendLine(sb, "      remote                connectCount    disconnectCount connections     send         receive");
+    appendLine(sb, "      remote                connectCount    disconnectCount connections     send(B)      receive(B)");
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/ClientEndpointsLogPublisher.java]: Bps?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/NetMeter.java [hunk 3]
```diff
@@ -56,32 +60,55 @@
 
     private Id receiveId;
 
+    private Id sendPacketId;
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/NetMeter.java]: same to cpu
logic of send and receive is same, except value of id/read from index is different, extract to one class,and InterfaceInfo  have 2 field of it.

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java [hunk 4]
```diff
@@ -161,8 +166,14 @@ private void printNetLog(StringBuilder sb, MeasurementNode osNode) {
...
   private void printCpuLog(StringBuilder sb, MeasurementNode osNode) {
     MeasurementNode cpuNode = osNode.findChild(OsMeter.OS_TYPE_CPU);
-    if (cpuNode != null && !cpuNode.getMeasurements().isEmpty()) {
-      appendLine(sb, "  cpu: %.2f%%", cpuNode.summary() * 100);
+    if (cpuNode == null || cpuNode.getMeasurements().isEmpty()) {
+      return;
+    }
+    double allRate = cpuNode.findChild(CpuMeter.TAG_All.value()).summary();
+    double processRate = cpuNode.findChild(CpuMeter.TAG_CURRENT.value()).summary();
+    if (allRate != 0 || processRate != 0) {
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java]: always print cpu usage.

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java [hunk 5]
```diff
@@ -16,92 +16,59 @@
...
-    return lastTotalTime;
+  public void calcMeasurements(List<Measurement> measurements, long msNow) {
+    update();
+    measurements.add(new Measurement(allCpuUsage.getId(), msNow, allCpuUsage.getUsage()));
+    measurements.add(new Measurement(processCpuUsage.getId(), msNow, processCpuUsage.getUsage()));
   }
 
-  public long getLastIdleTime() {
-    return lastIdleTime;
+  @VisibleForTesting
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: why so many VisibleForTesting?
just check measurements is not enough?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java [hunk 6]
```diff
@@ -16,92 +16,59 @@
...
+import com.netflix.spectator.api.Tag;
 
 public class CpuMeter {
-  private static final Logger LOGGER = LoggerFactory.getLogger(CpuMeter.class);
 
-  private double rate;
+  public static final Tag TAG_All = new BasicTag(OsMeter.OS_TYPE, OsMeter.OS_TYPE_ALL_CPU);
 
-  private long lastTotalTime;
+  public static final Tag TAG_CURRENT = new BasicTag(OsMeter.OS_TYPE, OsMeter.OS_TYPE_PROCESS_CPU);
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: why named process cpu and current at the same time?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/NetMeter.java [hunk 7]
```diff
@@ -43,101 +46,48 @@
...
   }
 
   public void calcMeasurements(List<Measurement> measurements, long msNow, long secondInterval) {
     refreshNet(secondInterval);
 
-    for (InterfaceInfo interfaceInfo : interfaceInfoMap.values()) {
-      measurements.add(new Measurement(interfaceInfo.sendId, msNow, interfaceInfo.sendRate));
-      measurements.add(new Measurement(interfaceInfo.receiveId, msNow, interfaceInfo.receiveRate));
-    }
+    interfaceUsageMap.values().stream()
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/NetMeter.java]: so complex, create a temp list and loop it to add to another list?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    return total;
+  }
+
+  @Override
+  protected long readCurrentBusyTime(String[] stats) {
+    return currentTotalTime - Long.parseLong(stats[4]);
+  }
+
+  @VisibleForTesting
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java]: too many VisibleForTesting

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/net/NetStat.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+  private Id id;
+
+  // send/recv bytes/packets
+  private long lastValue;
+
+  // Bps/pps
+  private double rate;
+
+
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/net/NetStat.java]: why always have useless empty lines?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java (hunk 8 above)
Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java]: ok, I will delete it

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java (hunk 6 above)
Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java]: ok. I will modify it

**Pull request merged**